### PR TITLE
[Refactor] refactor cloud native pk index's IndexValueWithVerPB

### PR DESCRIPTION
**Pull request by luohaha**
## Why I'm doing:
Field `values` in `IndexValueWithVerPB` contains two parts: 1) rowset id & segment id (short as `rssid`) 2) `rowid`. And in the future, we will support bulk load optimization. In this case, we don't need to set the `rssid` in `IndexValueWithVerPB`. 

## What I'm doing:
Splitting values into two parts (`rssid` and `rowid`) allows us to optimize the storage space by only use the `rowid` in some cases.

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [x] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/storage/lake/persistent_index_sstable.cpp [hunk 1]
```diff
@@ -49,8 +50,10 @@ Status PersistentIndexSstable::build_sstable(
     for (const auto& [k, v] : map) {
         IndexValueWithVerPB index_value_pb;
         for (const auto& index_value_with_ver : v) {
-            index_value_pb.add_versions(index_value_with_ver.first);
-            index_value_pb.add_values(index_value_with_ver.second.get_value());
+            auto* item = index_value_pb.add_items();
+            item->set_version(index_value_with_ver.first);
```
Comment (TszKitLo40) [inline be/src/storage/lake/persistent_index_sstable.cpp:54]: In `types.proto`, the comments said that verison will be set if MVCC is supported. So should version be set now?

Comment (luohaha) [reply be/src/storage/lake/persistent_index_sstable.cpp:54]: will removed in next pr

**Review by TszKitLo40 — APPROVED** [review]

File: gensrc/proto/types.proto [hunk 2]
```diff
@@ -171,9 +171,16 @@ message IsNullPredicatePB {
...
+    optional int64 version = 1;
+    // rowset id & segment id
+    optional uint32 rssid = 2;
+    optional uint32 rowid = 3;
+}
+
 message IndexValueWithVerPB {
-    repeated int64 versions = 1;
-    repeated uint64 values = 2;
+    repeated IndexValueWithVerItemPB items = 1;
```
Comment (wyb) [inline gensrc/proto/types.proto:183]:
```
IndexValueWithVerPB
IndexValuesWithVerPB
```

Comment (luohaha) [reply gensrc/proto/types.proto]: Done

**Review by wyb — APPROVED** [review]

File: gensrc/proto/types.proto [hunk 3]
```diff
@@ -172,8 +172,15 @@ message IsNullPredicatePB {
 }
 
 message IndexValueWithVerPB {
-    repeated int64 versions = 1;
-    repeated uint64 values = 2;
+    // will be set after mvcc support
```
Comment (kevincai) [inline gensrc/proto/types.proto:175]: will it cause any problem if upgrading from previous version?

looks like this protobuf change is not compatible at all.

Comment (luohaha) [reply gensrc/proto/types.proto:175]: Cloud native index is not released yet, I need to merge this PR as soon as possible 😆

Comment (kevincai) [reply gensrc/proto/types.proto:175]: ok

**Review by kevincai — APPROVED** [review]

**Comment by luohaha** [comment]
https://github.com/Mergifyio backport branch-3.3

**Pull request merged**